src/animation/animation_spritesheet.rs

```diff
@@ -62,27 +62,20 @@ fn apply_animation_player_spritesheet(
             elapsed += animation_clip.duration();
         }
 
-        animation.finished = false;
-
         let index = match animation_clip
             .keyframe_timestamps()
             .binary_search_by(|probe| probe.partial_cmp(&elapsed).unwrap())
         {
-            Ok(0) => 0, // this will probably the first frame in the paused state
-            Ok(n) if n >= animation_clip.keyframe_timestamps().len() - 1 => {
-                animation.finished = true;
-                return;
-            }
+            Ok(0) => 0, // this is probably the first frame in the paused state
+            Ok(n) if n >= animation_clip.keyframe_timestamps().len() - 1 => return,
             Ok(i) => i,
             Err(0) => return, // this clip isn't started yet
-            Err(n) if n > animation_clip.keyframe_timestamps().len() => {
-                animation.finished = true;
-                return;
-            }
+            Err(n) if n > animation_clip.keyframe_timestamps().len() => return,
             Err(i) => i - 1,
         };
 
         let keyframes = animation_clip.keyframes();
+        animation.finished = index == keyframes.len() - 1;
```

Comment:
That looks so weird, I think it would read better with parentheses.
But hey, if clippy wants it this way, we are going to do it this way.

Reply:
haha true, the `= index ==` reads a little weird. Clippy has some weird preferences with general `if` statements in my opinion but oh well, who am I to judge clippy.